internal/providers/vmware/vmware_amd64.go

```diff
@@ -68,26 +68,26 @@ func fetchRawConfig(f resource.Fetcher) (config, error) {
 		}
 
 		if _, ok := env.Properties["guestinfo.coreos.config.data"]; ok {
-			ovfDataKey = "guestinfo.coreos.config.data"
-			ovfEncodingKey = "guestinfo.coreos.config.data.encoding"
+			ovfDataKey = "coreos.config.data"
```

Comment:
Rather than having this `if` block, what about setting `ovfDataKey` & `ovfEncodingKey` to the `coreos.*` variables as defaults and updating them to `ignition.*` variables in the case that `guestinfo.ignition.config.data` is present in `env.Properties`. Right now if neither are the case we're relying on `info.String` to return an error when we send it `guestinfo.`